### DISCUSSION TITLE
Bridging IOL

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1477] [Show and tell]
I succeeded to make a dumb bridge on a IOL router. This one just bridges 2 PCs. I used bridge domains instead of the old bridge groups. Ill try to push the config to more complex scenarios in the next days.  If they work, they ll come to netlab later.  Relevant config is here:  

https://your.ls/h30yi

**Comment by ipspace** [comment]
Nice catch ;) Copied vlan/csr.j2 into vlan/iol.j2, tweaked device settings, and the simplest test case succeeded. Running tests... Trunks work ;)) Looking good...

**Reply by DanPartelly** [reply]
wow man, you are quick 🚀 🚀 🚀 
> Nice catch ;) Copied vlan/csr.j2 into vlan/iol.j2, tweaked device settings, and the simplest test case succeeded. Running tests... Trunks work ;)) Looking good...

**Comment by ipspace** [comment]
Of course something had to fail. Routed subinterfaces were added to the bridge domains, so tests [56]* were failing. Fixed that, all tests passed apart from 63* but that's a netlab limitation. Updated the caveats and VLAN module documentation.

Thanks again for figuring this out!

**Reply by DanPartelly** [reply]
Excellent! Thanks for all this fast work. For after 1.9.2, Ill work on enabling STP and lagg modules on various Ciscos. then need to learn more about netlab core so I can work on modules. 

> Of course something had to fail. Routed subinterfaces were added to the bridge domains, so tests [56]* were failing. Fixed that, all tests passed apart from 63* but that's a netlab limitation. Updated the caveats and VLAN module documentation.
> 
> Thanks again for figuring this out!

**Reply by ipspace** [reply]
> For after 1.9.2, Ill work on enabling STP and lagg modules on various Ciscos

👍

> then need to learn more about netlab core so I can work on modules.

Any idea what you'd like to add?

**Reply by DanPartelly** [reply]
Netlab already has the most interesting stuff.  Maybe enhanced object tracking and SLAs. Not sure it qualifies as a module. I know Cisco and Junpier both have them, but i don't know about other vendors. Is there anything you had in mind but never had time to do ?
> Any idea what you'd like to add?

**Reply by jbemmel** [reply]
MC-LAG support

**Reply by DanPartelly** [reply]
> MC-LAG support

Ill look to see of any of our supported devices can do multichassis. You know any ? (images, not real hw)

**Comment by jbemmel** [comment]
I saw a note about vlans 1002-5 being reserved on some platforms - is that modeled and checked?

if not, that would be something to add - gets annoying quickly

**Reply by DanPartelly** [reply]
Its Cisco switches, and I do not know how long they gonna carry this debt along. Historically they used those for bridging fddi and token ring. It is specified as a caveat, not modeled. 

1002 fddi  101002     1500  -      -      -        -    -        0      0 
1003 tr    101003     1500  -      -      -        -    -        0      0   
1004 fdnet 101004     1500  -      -      -        ieee -        0      0   
1005 trnet 101005     1500  -      -      -        ibm  -        0      0   

> I saw a note about vlans 1002-5 being reserved on some platforms - is that modeled and checked?
> 
> if not, that would be something to add - gets annoying quickly

**Reply by jbemmel** [reply]
We have quirk modules for that - e.g. look at Dell OS10 which doesn’t support OSPF on SVI interfaces

Something similar could be done for IOS - but since reserved vlan ranges are common, we may want to build something more generic (e.g. reserved_vlan_ranges property in vlan module)

**Reply by ipspace** [reply]
> We have quirk modules for that - e.g. look at Dell OS10 which doesn’t support OSPF on SVI interfaces

👆

> Something similar could be done for IOS - but since reserved vlan ranges are common

I only encountered it on IOS. What other platforms have similar limitations?

**Reply by jbemmel** [reply]
e.g. Dell OS10 reserves 4094-4096 (https://www.dell.com/support/kbdoc/en-us/000125788/dell-emc-networking-configuring-vlan-on-switches)

Not that 4096 could be represented in 12 bits but that’s what they say. I expect there may be others - the notion of a “reserved vlan range” is common enough

**Reply by DanPartelly** [reply]
Actually it is modeled. I forgot that I simlinked  the iosvl2 quirk module as quirks for ioll2. 
> We have quirk modules for that - e.g. look at Dell OS10 which doesn’t support OSPF on SVI interfaces
>

**Reply by ipspace** [reply]
Isn't it nice when you start working on something just to discover you already did it ;))

**Comment by ipspace** [comment]
IOL/IOLL2 work ;)
